packages/@react-aria/interactions/src/useMove.ts

```diff
@@ -172,41 +173,61 @@ export function useMove(props: MoveEvents): MoveResult {
       };
     }
 
-    let triggerKeyboardMove = (deltaX: number, deltaY: number) => {
+    let triggerKeyboardMove = (deltaX: number, deltaY: number, isPage?: boolean) => {
       start();
-      move('keyboard', deltaX, deltaY);
+      move('keyboard', deltaX, deltaY, isPage);
       end('keyboard');
     };
 
     moveProps.onKeyDown = (e) => {
-      // don't want useMove to handle shift key + arrow events because it doesn't do anything
-      if (e.shiftKey) {
-        return;
-      }
       switch (e.key) {
         case 'Left':
         case 'ArrowLeft':
           e.preventDefault();
           e.stopPropagation();
-          triggerKeyboardMove(-1, 0);
+          triggerKeyboardMove(-1, 0, e.shiftKey);
           break;
         case 'Right':
         case 'ArrowRight':
           e.preventDefault();
           e.stopPropagation();
-          triggerKeyboardMove(1, 0);
+          triggerKeyboardMove(1, 0, e.shiftKey);
           break;
         case 'Up':
         case 'ArrowUp':
           e.preventDefault();
           e.stopPropagation();
-          triggerKeyboardMove(0, -1);
+          triggerKeyboardMove(0, -1, e.shiftKey);
           break;
         case 'Down':
         case 'ArrowDown':
           e.preventDefault();
           e.stopPropagation();
-          triggerKeyboardMove(0, 1);
+          triggerKeyboardMove(0, 1, e.shiftKey);
+          break;
+          // TODO: include page size as an option? these are page movements, also Shift + arrows count for that
+          // it can be different depending on the slider/s 2D sliders home/end&pageup/down are on two axis
+          // in 1D sliders, home/end go to min/max, vs pageup/down are page size increments
+          // should triggerKeyboardMove include an is PageMove? include a key?
```

Comment:
https://github.com/adobe/react-spectrum/pull/2491/files/13132ed1d7eb3ca48af4d3a78467b8defb1537e9#r736024259

@LFDanLu LFDanLu 7 days ago Member
Related to my useColorWheelState sentiment, IMO adding page size as an option would be nice, but perhaps we can wait till people request it. IMO I think we are fine w/ omitting pageMove/the keyboard key from triggerKeyboardMove for now, did you have any use cases in mind?

 
@snowystinger snowystinger 7 days ago Author Member
yeah, given that it's really stately that cares about the size, I've been keeping it over there instead of here or in the aria hooks where useMove is used